ui/webui/src/components/time/DateAndTime.jsx

```diff
@@ -0,0 +1,490 @@
+              // update the list of cities based on the selected region
+              // TODO: make this dynamic in the CitySelect itself, based on selected region ?
+              setRegionCities(timezones[selection]);
+              // remember it locally so we can show it
+              setCurrentSelection(selection);
+              // then close the Select
+              setIsOpenRegionSelect(false);
+          }}
+          isOpen={isOpenRegionSelect}
+          variant={SelectVariant.typeahead}
+          placeholderText={_("Select region")}
+          selections={currentSelection}
+        >
+            {regions.map(region => <SelectOption
+              key={region.value}
+              value={region} />)}
+        </Select>
+    );
+};
+
+const CitySelect = ({ timezones, regionCities, selectedCity, setSelectedCity }) => {
+    const [isOpenCitySelect, setIsOpenCitySelect] = useState(false);
+    const [currentSelection, setCurrentSelection] = useState(null);
+    return (
+        <Select
+          onToggle={(_, isOpen) => setIsOpenCitySelect(isOpen)}
+          onSelect={(e, selection) => {
+              // set the value from the Select
+              setSelectedCity(selection);
+              // remember it locally so we can show it
+              setCurrentSelection(selection);
+              // then close the Select
+              setIsOpenCitySelect(false);
+          }}
+          isOpen={isOpenCitySelect}
+          variant={SelectVariant.typeahead}
+          placeholderText={_("City")}
+          selections={currentSelection}
+        >
+            {}
+            {regionCities.map(city => <SelectOption
+              key={city.value}
+              value={city} />)}
+        </Select>
+    );
+};
+
+const EditSettingsCard = ({
+    idPrefix,
+    setEditModeEnabled,
+    automaticTimeEnabled,
+    setAutomaticTimeEnabled,
+    timezones,
+    regions,
+    selectedRegion,
+    setSelectedRegion,
+    regionSelected,
+    regionCities,
+    setRegionCities,
+    selectedCity,
+    setSelectedCity,
+    systemDateTime,
+    setSystemDateTime,
+}) => {
+    const [selectedDate, setSelectedDate] = useState(null);
+    const [selectedTime, setSelectedTime] = useState(null);
+    return (
+        <Card>
+            <CardBody>
+                <Flex direction={{ default: "column" }} spacer={{ default: "spaceItemsXs" }}>
+                    <FlexItem>
+                        <Text className="time-edit-heading">
+                            {_("Timezone")}
+                        </Text>
+                    </FlexItem>
+                    <FlexItem spacer={{ default: "spacerXs" }} />
+                    <FlexItem>
+                        <Flex direction={{ default: "row" }} alignSelf={{ default: "alignItemsCenter" }}>
+                            <Flex direction={{ default: "column" }}>
+                                <FlexItem>
+                                    <Text className="time-edit-subheading">
```

Comment:
This should be all reimplemented to be a form. See https://www.patternfly.org/v4/components/form/#grid-form

Reply:
The code is using the form layout now. :)